lib/rules/no-duplicate-landmark-labels.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const Rule = require('./base');
+const AstNodeInfo = require('../helpers/ast-node-info');
+
+function createErrorMessage(element) {
+  return `Duplicate label on <${element}> detected. If two of the same landmark elements or landmark roles are found in the same template, ensure that they have a unique label (provided by the aria-label or aria-labelledby attribute).`;
+}
+
+const ROLE_LANDMARK_MAP = {
+  banner: 'header',
+  main: 'main',
+  complementary: 'aside',
+  form: 'form',
+  search: 'form',
+  navigation: 'nav',
+  contentinfo: 'footer',
+};
+
+const LANDMARK_ELEMENTS = ['header', 'main', 'aside', 'form', 'main', 'nav', 'footer'];
+
+const LANDMARK_ROLES = Object.keys(ROLE_LANDMARK_MAP);
+
+module.exports = class NoDuplicateLandmarkLabels extends Rule {
+  constructor(options) {
+    super(options);
+    this._landmarkHash = [];
+  }
```

Comment:
I tried with `Set` but since the object references change every time, it's not able to catch the duplicate it seems.
```js
  constructor(options) {
    super(options);
    this._landmarkHash = new Set();
  }
  visitor() {
    return {
      ElementNode(node) {
        ...
        if (isLandMarkElement || isLandMarkRole) {
          // check for label
          const label = AstNodeInfo.elementAttributeValue(node, 'aria-label') || '';
          const landmark = isLandMarkElement ? node.tag : ROLE_LANDMARK_MAP[roleValue];
          let current = { landmark, label };
          if (this._landmarkHash.has(current)) {
            this.log({
              message: createErrorMessage(node.tag),
              line: node.loc && node.loc.start.line,
              column: node.loc && node.loc.start.column,
              source: this.sourceForNode(node),
            });
          } else {
            this._landmarkHash.add(current);
          }
...
 ```

Reply:
Ya, makes sense, sorry for the troubles.